File: 62f.js
const fs = require('fs');
const MOD = 1000000007;

function sumOfTripleProducts() {
	const fileContent = fs.readFileSync('input.txt', 'utf8');
	const lines = fileContent.toString().split('\n');

	if (lines.length > 1) {
		const n = Number(lines[0].trim());
		const a = lines[1].split(' ').map(Number);

		let totalSum = 0;
		let totalProductSum = 0;

		// Вычисляем общую сумму
		for (let num of a) {
			totalSum = (totalSum + num) % MOD;
		}

		// Вычисляем сумму произведений
		for (let i = 0; i < n; i++) {
			let current = a[i];

			// Сумма оставшихся элементов
			let sumAfterCurrent = (totalSum - current + MOD) % MOD;
			let countAfterCurrent = n - (i + 1); // Количество элементов после текущего

			// Умножаем текущее число на сумму остальных
			totalProductSum = (totalProductSum + (current * sumAfterCurrent % MOD) * countAfterCurrent % MOD) % MOD;
		}

		// Делим на 6, так как каждое произведение считается трижды
		totalProductSum = (totalProductSum * modInverse(6, MOD)) % MOD;

		return totalProductSum;
	}
	return 0;
}

// Функция для нахождения обратного элемента по модулю
function modInverse(a, m) {
	let m0 = m, t, q;
	let x0 = 0, x1 = 1;

	if (m === 1) return 0;

	while (a > 1) {
		q = Math.floor(a / m);
		t = m;

		m = a % m;
		a = t;
		t = x0;

		x0 = x1 - q * x0;
		x1 = t;
	}

	if (x1 < 0) x1 += m0;

	return x1;
}

// Записываем результат в файл
fs.writeFileSync('output.txt', sumOfTripleProducts().toString());

File: 62f.py
MOD = 1000000007

# Функция для нахождения модульного обратного числа
def mod_inverse(a, p):
    return pow(a, p - 2, p)

def main():
    # Чтение данных
    with open('input.txt', 'r') as file:
        n = int(file.readline().strip())
        arr = list(map(int, file.readline().strip().split()))

    S_total = 0
    S_squares = 0
    S_cubes = 0

    for num in arr:
        S_total = (S_total + num) % MOD
        S_squares = (S_squares + num * num) % MOD
        S_cubes = (S_cubes + num * num * num) % MOD

    # Используем формулу для подсчета суммы произведений
    totalSum = (S_total * S_total % MOD * S_total % MOD - 3 * S_total % MOD * S_squares % MOD + 2 * S_cubes % MOD) % MOD

    # Корректируем, чтобы избежать отрицательного значения
    if totalSum < 0:
        totalSum += MOD

    # Подсчитываем общее количество тройных произведений
    inv6 = mod_inverse(6, MOD)
    result = (totalSum * inv6) % MOD

    # Запись результата в файл
    with open('output.txt', 'w') as file:
        file.write(str(result) + '\n')

if __name__ == "__main__":
    main()
